src/core/mount.c

```diff
@@ -1484,7 +1487,48 @@ static void mount_sigchld_event(Unit *u, pid_t pid, int code, int status) {
          * race, let's explicitly scan /proc/self/mountinfo before we start processing /usr/bin/(u)mount
          * dying. It's ugly, but it makes our ordering systematic again, and makes sure we always see
          * /proc/self/mountinfo changes before our mount/umount exits. */
-        (void) mount_process_proc_self_mountinfo(u->manager);
+        if (!mnt_has_fsinfo())
+                (void) mount_process_proc_self_mountinfo(u->manager);
+        else {
+                struct libmnt_fs *fs;
+                const char *where;
+                int mnt_id;
+
+                fs = mnt_new_fs();
+                if (!fs) {
+                        log_error_errno(-ENOMEM, "failed to allocate libmount fs struct");
```

Comment:
log_oom()

Reply:
Yes, this was another thing I wasn't sure about.

I thought maybe log_oom() was meant to be used when you encounter a situation where you need to limit possible damage due to an error that shouldn't happen and could have significant adverse consequence continuing.

It sounds like my impression is correct.